### ISSUE TITLE
Improve al-repeat directive: remove orderBy and extract "store to"

### ISSUE THREAD

**Issue by dev-rke** [lega911/angular-light#112]
Hi,

TLDR: i recommend to remove the implicit usage of the orderBy-Filter and extract the store-to functionality of the repeat directive into an own filter.

I recommend to remove the orderBy statement from the al-repeat directive.
Sometimes you want to filter first, then order it. 
Sometimes you want to order first and then filter it.
Sometimes you want to filter first, order second and filter third.
There can be different filter results, e.g. if you have dependencies between the items of an array or object.
This would make the al-repeat directive more clear and should not affect performance.
The existing orderBy filter fulfills all requirements, so the implicit orderBy statement in al-repeat directive is redundant.

Also the store-to functionality of the repeat directive should be extracted into an own filter, i wrote my own implementation which i would recommend to include as an official filter:

``` coffeescript
# filter to store data into a variable
alight.filters.storeTo = (exp, cd, env) ->
   onChange: (input) ->
      env.setValue input
      cd.setValue exp, input
      cd.scan()
```

I recommend to extract the store-to functionality in an own filter, because in a chain of filters it is not possible to save an array of results between two filters.

Why is this more flexible?
When you try to display only 10 items of an array of thousands of items, but want to show a pagination for the filtered data, you might want to slice the items after you filtered it.
Example:

``` html
<div al-repeat="item in items | filter:filterParams | storeTo:filteredItems | orderBy:'name',true | slice:slice.min,slice.max">
  {{item.name}}
</div>
<ul>
  <li al-repeat="filteredItems">
    {{$index}}
  </li>
</ul>
```

Using the existing implementation of the store-to functionality would save the sliced data, so there is no option to get all filtered data.

**Comment by lega911** [comment]
storeTo was done, example http://jsfiddle.net/lega911/zrmnj67b/

orderBy is an external filter (it's not part of al-repeat), but al-repeat uses it for objects, when input value - is an object, in this case you can't use external filter orderBy, because orderBy transforms arrays only.

For arrays you can use |orderBy, as you want.

**Comment by dev-rke** [comment]
Cool, thank you. :-)
